vamp lurker strain

Diegoflores31 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->
Adds a new strain for lurker (vampire is the name for now but i can be changed i just lack the creativity) has less health than the average lurker but its faster and slashes faster but deals less damage .

**### BASE STATS**
health : 390
armor : 20 
slash damage : 35
speed : 0.1 faster than base lurker   // for reference cloaked speed is 0.2
slash speed : 2


Vamp lurker cannot cloak but has a larger kit of abilities focused on dealing damage and healing making it a high risk high reward backliner with skill based abilities rather than just stun.
### **Abilities :**

**Rush:** 
Shorter version of pounce (4 tiles) instead of stunning it will daze and slightly screenshake the target .
damage : 30
cooldown : 6 seconds and 3 if you land it.

**Flurry:**
AOE attack that deals damage to the targets in front of you in a 1x3 area . if landed it will heal you by the same amount and apply a small slow for the user ( very short second slow)
damage: 30
heal : 30
cooldown : 3 seconds if missed 

**Tail Jab:**
Targeted attack will deal a small amount of damage to the target and knock him away from you . if you use it on a target in critical state it will execute it healing you a LOT.
damage : 30
Execution damage : 80 with penetration
cooldown : 7 seconds 
heal : 150
critical state :  this occurs when the target either paincrits or falls INCONCIOUS

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
Lurker lacks strains and i looked up in the Trello that Lurker strain was required . i tried to follow the indicators but kinda ended up with something else i guess but i really like how it ended so i am making this PR to see what you think about it.


## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl: diegoflores31
add: Added a new strain for lurker "Vampire".
add: Vampire Lurker exchanges all of your abilities for a fast paced combat style more focused into dealing damage and 
mobility.
add: Active 1 : Rush . Pounces for a maximun of 4 tiles and slashes the objetive once on impact . applying a screenshake and daze to the target . Landing this ability reduces the cooldown by half. (cooldown 6 seconds)
add: Active 2 . Flurry : unleashes a 1X3 slash at the targeted direction that slows your enemies on impact healing you by 30 hp . (cooldown 3 seconds)
add: Active 3 : Tail Jab : Attacks your enemies from a maximun of  2 tiles away while dealing a small amount of damage ( 20)  and knocking them down . if you attack a enemy that is on critical state it will deal 80 damage with penetration and heal you by 150 hp. (cooldown 7 seconds)
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Waseemq1235 commented:
draft this if it aint complete

also, lurker's pounce is kinda the only way they can deal with shotgun cancer, and this strain is just gonna be instantly dead if it encounters one. I do like the idea of stun-less combat, but a stunless caste against a stun-equipped marine simply wont work, unless you make this strain immune to stuns (cursed). base lurker is already flimsy as fuck against shotguns, this just scales the issue up

Carnage068 commented:
YES

Carnage068 commented:
> also, lurker's pounce is kinda the only way they can deal with shotgun cancer, and this strain is just gonna be instantly dead if it encounters one. I do like the idea of stun-less combat, but a stunless caste against a stun-equipped marine simply wont work, unless you make this strain immune to stuns (cursed). base lurker is already flimsy as fuck against shotguns, this just scales the issue up

Also easy solution to this would be to just give it a much shorter pounce duration. Like, maybe half of runner pounce, while still applying the daze and screenshake. Also I feel like Lurker with even less health might be a little too squishy, lol. I don't think everything here balances out super well, and the strain if released in its current state would probably end up being considerably underpowered, if not downright awful. Also is it going to be using plasma or nah?

Waseemq1235 commented:
> > also, lurker's pounce is kinda the only way they can deal with shotgun cancer, and this strain is just gonna be instantly dead if it encounters one. I do like the idea of stun-less combat, but a stunless caste against a stun-equipped marine simply wont work, unless you make this strain immune to stuns (cursed). base lurker is already flimsy as fuck against shotguns, this just scales the issue up
> 
> Also easy solution to this would be to just give it a much shorter pounce duration. Like, maybe half of runner pounce, while still applying the daze and screenshake. Also I feel like Lurker with even less health might be a little too squishy, lol. I don't think everything here balances out super well, and the strain if released in its current state would probably end up being considerably underpowered, if not downright awful. Also is it going to be using plasma or nah?


with the current slug meta even a short pounce just won't do, especially with the extra squishiness of the strain. half runner pounce is literally enough to get a single slash in before you get PBd and ended on the spot, but yeah, if this gets released, then this is gonna be even worse than boxer

Carnage068 commented:
> with the current slug meta even a short pounce just won't do, especially with the extra squishiness of the strain. half runner pounce is literally enough to get a single slash in before you get PBd and ended on the spot, but yeah, if this gets released, then this is gonna be even worse than boxer

Well, what do you want them to do? From what I've seen, base Lurker pounce isn't just enough for them to 'deal' with shotgunners, it's enough for them to put them in pain crit without even being able to get up. Giving Vampire Lurker the ability to do the same would just be fucking stupid, since unlike Base they can heal themselves, plus they have higher movement speed, so they're better at avoiding gunfire than regular Lurker.

I do think that lowering their health (which is already somewhat low at 450 health) is kind of stupid though. Kind of takes away from them supposedly having more survivability than base.

Carnage068 commented:
> with the current slug meta even a short pounce just won't do, especially with the extra squishiness of the strain. half runner pounce is literally enough to get a single slash in before you get PBd and ended on the spot, but yeah, if this gets released, then this is gonna be even worse than boxer

Flores just messaged me on Discord, told me he forgot to mention in PR that he gave it armor, which he did. 20 armor, to be exact, and even though it has a whole 100 less health, it will still take significantly less damage overall from shotguns. Also, the 100 is likely to be changed to 60 or 50 less health instead. By that point, I honestly don't think it will need to have a stun at all.

Waseemq1235 commented:
> > with the current slug meta even a short pounce just won't do, especially with the extra squishiness of the strain. half runner pounce is literally enough to get a single slash in before you get PBd and ended on the spot, but yeah, if this gets released, then this is gonna be even worse than boxer
> 
> Flores just messaged me on Discord, told me he forgot to mention in PR that he gave it armor, which he did. 20 armor, to be exact, and even though it has a whole 100 less health, it will still take significantly less damage overall from shotguns. Also, the 100 is likely to be changed to 60 or 50 less health instead. By that point, I honestly don't think it will need to have a stun at all.

seems ok then, but cursed as fuck, this is like a ravager but on a T2 scale that nukes marines, i like it.

Carnage068 commented:
> seems ok then, but cursed as fuck, this is like a ravager but on a T2 scale that nukes marines, i like it.

On a T2 scale, yeah. 400 health and 20 armor compared to Base Rav's 650 health and 25 armor seems pretty fair to me. Honestly I can see myself playing this a good amount of the time- Sounds way more fun than Base Lurker for me.

Blithering commented:
One way to maybe have it not die to the first shotgunner it sees might be to have draining health when you're already at full give a temporary stun resistance buff? 
So if you get the bite in, you can avoid getting wombo comboed by the retort, but need to be sure a SMG doesn't wing you on the way in.

Carnage068 commented:
> One way to maybe have it not die to the first shotgunner it sees might be to have draining health when you're already at full give a temporary stun resistance buff? So if you get the bite in, you can avoid getting wombo comboed by the retort, but need to be sure a SMG doesn't wing you on the way in.

That would honestly be a really damn good (good as in balanced, not op, lmao) way of having Vamp Lurkers being able to take on shotgunners.

Diegoflores31 commented:
> One way to maybe have it not die to the first shotgunner it sees might be to have draining health when you're already at full give a temporary stun resistance buff? So if you get the bite in, you can avoid getting wombo comboed by the retort, but need to be sure a SMG doesn't wing you on the way in.

it has armor and i will tweak it to 390 or something so it literally wont die on a single PB

Diegoflores31 commented:
**UPDATE :::** 
health values increased to 390 (tested ingame can survive 3 PBS with buckshot)
Added Strain flavor text 
tweaked penetration for Tail jab Execution
Tweaked cooldowns on all abilities
Tweaked damage values 
Added Macro paths for all abilities

PhantomEpicness commented:
UNFATHOMABLY BASED 

_Disclaimer_

- CM-13 is not related to the Disney owned "Aliens" Franchise and IP This is an entirely fan-made project. In case of an investigation by any federal entity or similar, I do not have any involvement with this group or with the people in it, I do not know how I am here, probably added by a thrid party, I do not support any actions by the member of this group.

Carnage068 commented:
Thank you for putting in the actual numbers, makes it a lot prettier to look at. Also gives a better idea of how it is balance-wise.

stanalbatross commented:
if this functions in localtesting it will be testmerged

Carnage068 commented:
> if this functions in localtesting it will be testmerged

Holy shit if this ends up getting testmerged within the next few days then we did not have to wait long for it at all, LOL. 5-7 days for a PR to go through to testing is crazy.

Chainsawicus commented:
> > if this functions in localtesting it will be testmerged
> 
> Holy shit if this ends up getting testmerged within the next few days then we did not have to wait long for it at all, LOL. 5-7 days for a PR to go through to testing is crazy.

It's because it's an approved feature on the Trello.

https://trello.com/c/uVRm7PC5/6-lurker-xeno-strain

Carnage068 commented:
> > > if this functions in localtesting it will be testmerged
> > 
> > 
> > Holy shit if this ends up getting testmerged within the next few days then we did not have to wait long for it at all, LOL. 5-7 days for a PR to go through to testing is crazy.
> 
> It's because it's an approved feature on the Trello.
> 
> https://trello.com/c/uVRm7PC5/6-lurker-xeno-strain

Yeye I know that it was on the Trello. I just didn't know that Trello stuff is so quick to get merged/testmerged. I guess it makes sense though.

stanalbatross reviewed (changes requested):
![image](https://user-images.githubusercontent.com/66756236/188210460-88d44920-8999-4560-addc-1ae7a138a58d.png)
better desc needed
better changelog too
![image](https://user-images.githubusercontent.com/66756236/188210714-692fa850-a3ad-407d-b622-6f661a96caa7.png)
comma and full stop

Rushing at a lying down enemy does nothing (odd behaviour)

Nanu308 commented:
Big fan of seeing this coming to fruition, see stan's comment then its tm time.

Diegoflores31 commented:
> Rushing at a lying down enemy does nothing (odd behaviour)

tested ingame . rushing at lying down enemies actually works but seems like the Pounce system refuses to apply effects at targets that are lying down AND in critted for some reason . i dont really know how to fix this issue in particular since its something that comes from pouncing or even from somewhere else .

Diegoflores31 commented:
**UPDATE :::**
-Increased damage on all abilities to 30 to be more consistent with the slash damage
-Removed self stun on lunge ( vamp lurker has no stun and keeping this would make it too easy to PB)
-Corrected text as requested by review
-Changed Strain description as requested by review

stanalbatross commented:
> > Rushing at a lying down enemy does nothing (odd behaviour)
> 
> tested ingame . rushing at lying down enemies actually works but seems like the Pounce system refuses to apply effects at targets that are unconcious . this is to facilitate caps i guess and i would like to keep it that way to allow lurkers to use the execution ability. but tell me if you want it gone and i can change it

alright sounds fine no need to change

usnpeepoo commented:
love the abilities but i feel like in practice it would just be melted before it could get close/try its abilities.

Carnage068 commented:
> Big fan of seeing this coming to fruition, see stan's comment then its tm time.

Vamp Lurker may very well become the most beloved strain in all of CM. I know I fucking adore it at least, and pretty much everyone else here and everyone I've talked to on the Discord seem to be all for it. Bloody boi has a bright future ahead of it, I am sure.

Agameofscones reviewed (approved):
sprites are OK

Diegoflores31 commented:
real

morrowwolf reviewed (changes requested):
Strain sprite is broken on live.
Hotkeys for abilities broken on live.

Diegoflores31 commented:
**UPDATE**
Tail jab now only executes Hardcrit/inconcious targets ( this avoids abuse on synths/preds)
Vamp Lurker Hotkeys now are usable (Xeno_primary_action , etc)

Diegoflores31 commented:
**Balance update**(requested by devs)
-reduced Rush damage to 20
-reduced flurry damage and heal to 20
-Flurry cooldown now its a flat 3 seconds
-You need to be adjacent to your target in order to EXECUTE them 
-Execute ability now has a 1 second charge time.
-Added attack text to better feedback sorrounding mobs about your actions.
-Reduced Tail Jab damage to 20

Shad0vvs commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -119,8 +119,8 @@
-	Xeno.visible_message(SPAN_XENOWARNING("[Xeno] slashes frantically !"), \
-	SPAN_XENOWARNING("You unleash a barrage of slashes!"))
+	Xeno.visible_message(SPAN_DANGER("[Xeno] slashes frantically the area in front of him!"), \
+		SPAN_XENOWARNING("You unleash a barrage of slashes!"))
```
Fix tab.

Shad0vvs commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -192,7 +192,17 @@
+			return
+		Xeno.visible_message(SPAN_DANGER("[Xeno] Prepares for devastating attack on [Target]."), \
+			SPAN_XENOWARNING("You carefully aim your tail towards [Target] vital organs."), null, 5)
+		if(!do_after(Xeno, 10, INTERRUPT_NO_NEEDHAND, BUSY_ICON_GENERIC))
```
Xeno abilities get BUSY_ICON_HOSTILE

Shad0vvs commented on code/modules/mob/living/carbon/xenomorph/mutators/strains/lurker/vampire.dm:
```diff
@@ -0,0 +1,38 @@
+/datum/xeno_mutator/Vampire
+	name = "STRAIN: Lurker - Vampire"
+	description = "You lose the ability to cloak in exchange for faster speed and armor, along with extra abilities that will allow you to bleed your opponents and execute them, granting you great healing. "
```
```suggestion
	description = "You lose the ability to cloak in exchange for faster speed and armor, along with extra abilities that will allow you to bleed your opponents and execute them, granting you great healing."
```

morrowwolf reviewed (changes requested):
Needs to uninvis lurkers when they switch strains or they remain perma invis.

Diegoflores31 commented:
> Needs to uninvis lurkers when they switch strains or they remain perma invis.

Fix comming in a few hours

Lumdor commented:
Some suggestions after playing as vamp lurker and against it. Flurry would feel better to use if it were an on-click ability instead of instant when you hit your macro. The tail jab having a two-tile range with the knockdown/knockback is also very strong and in my opinion, should have a one-tile range because you already have the gap closer to use it.

stanalbatross reviewed (changes requested):
- flurry needs to be aimable with mouse not based on mob dir (if anything change dir to mouse loc then call it)
- execute needs to have checks so it cannot be done through walls

morrowwolf reviewed (changes requested):
All of the abilities need attack logs.

Diegoflores31 commented:
**UPDATE**
increased heal and damage on flurry back to 30
increased execute heal to 150 from 100
made slash damage be a solid 35 instead of variable betwen 35 and 25
Added Restrictions so you are unable to hit thru walls with Tail jab 
Added very soul interaction with Tail jab hitting windows. ( its breaks them :O )
changed "swipe" sound from flurry to Miss instead of Hit 
Flurry is now an activable ability.
added logs to all Vamp lurker abilities

stanalbatross commented on code/modules/mob/living/carbon/xenomorph/Xenomorph.dm:
```diff
@@ -244,6 +244,7 @@
 	//////////////////////////////////////////////////////////////////
 	var/weedwalking_activated = 0 //Hivelord's weedwalking
 	var/tunnel = 0
+	var/invisible = 0 // for check on lurker invisibility
```
no more of these shit vars please

Diegoflores31 replied:
changed all of those for boolean and grouped them together

stanalbatross commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_macros.dm:
```diff
@@ -4,11 +4,32 @@
+/datum/action/xeno_action/verb/verb_rush()
+	set category = "Alien"
+	set name = "Rush"
+	set hidden = 1
+	var/action_name = "Rush"
+	handle_xeno_macro(src, action_name)
+
+/datum/action/xeno_action/verb/verb_flurry()
+	set category = "Alien"
+	set name = "Flurry"
+	set hidden = 1
+	var/action_name = "Flurry"
+	handle_xeno_macro(src, action_name)
+
+/datum/action/xeno_action/verb/verb_tail_jab()
+	set category = "Alien"
+	set name = "Tail Jab"
+	set hidden = 1
```
hidden is bool

stanalbatross commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -69,33 +67,188 @@
-	var/datum/behavior_delegate/lurker_base/behavior = xeno.behavior_delegate
-	if (istype(behavior))
-		behavior.next_slash_buffed = TRUE
+	var/datum/behavior_delegate/lurker_base/BD = Xeno.behavior_delegate
+	if (istype(BD))
+		BD.next_slash_buffed = TRUE
```
why

stanalbatross commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -37,30 +36,29 @@
-			if (istype(behavior))
-				behavior.on_invisibility_off()
+		if (Xeno.mutation_type == LURKER_NORMAL)
+			var/datum/behavior_delegate/lurker_base/BD = Xeno.behavior_delegate
+			if (istype(BD))
+				BD.on_invisibility_off()
 
```
why

stanalbatross commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -10,16 +10,15 @@
-		var/datum/behavior_delegate/lurker_base/behavior = xeno.behavior_delegate
-		behavior.on_invisibility()
+	if (Xeno.mutation_type == LURKER_NORMAL)
+		var/datum/behavior_delegate/lurker_base/BD = Xeno.behavior_delegate
+		BD.on_invisibility()
```
why

Diegoflores31 replied:
sorry mb i did this PR before that was changed and forgor to update that part

stanalbatross commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -69,33 +67,188 @@
+	if (!action_cooldown_check())
+		return
+
+	Xeno.visible_message(SPAN_DANGER("[Xeno] slashes frantically the area in front of him!"), \
```
him???????????

stanalbatross reviewed (changes requested):
your var names are all over the fucking place
- do not use single-letter or two-letter var names
- use snake_case in var names, DO NOT use capital letters
- do not change var names to make them worse

Diegoflores31 commented:
**UPDATE**
-changed some vars to Boolean as requested 
-corrected BD var to be "behaviour" as inteded
- Flurry text now call xeno IT instead of Him

morrowwolf reviewed (changes requested):
Strain is without icon.

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -69,33 +67,188 @@
+// VAMPIRE LURKER
+
+/datum/action/xeno_action/activable/pounce/rush/additional_effects(mob/living/L) //pounce effects
+	var/mob/living/carbon/Target = L
```
snake_case var name

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -69,33 +67,188 @@
+
+/datum/action/xeno_action/activable/pounce/rush/additional_effects(mob/living/L) //pounce effects
+	var/mob/living/carbon/Target = L
+	var/mob/living/carbon/Xenomorph/Xeno = owner
```
the snake case var name

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -69,33 +67,188 @@
+
+// VAMPIRE LURKER
+
+/datum/action/xeno_action/activable/pounce/rush/additional_effects(mob/living/L) //pounce effects
```
i would rather this was a descriptive var name but it's only used right next to declaration so... erm...

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -69,33 +67,188 @@
 /datum/action/xeno_action/onclick/lurker_assassinate/proc/unbuff_slash()
-	var/mob/living/carbon/Xenomorph/xeno = owner
-	if (!istype(xeno))
+	var/mob/living/carbon/Xenomorph/Xeno = owner
```
do not unsnakecase this

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -37,30 +36,29 @@
 
-/datum/action/xeno_action/onclick/lurker_assassinate/use_ability(atom/target)
-	var/mob/living/carbon/Xenomorph/xeno = owner
+/datum/action/xeno_action/onclick/lurker_assassinate/use_ability(atom/A)
```
no unsnakecasing

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -37,30 +36,29 @@
-		animate(xeno, alpha = initial(xeno.alpha), time = 0.1 SECONDS, easing = QUAD_EASING)
-		to_chat(xeno, SPAN_XENOHIGHDANGER("You feel your invisibility end!"))
-		button.icon_state = "template"
+	var/mob/living/carbon/Xenomorph/Xeno = owner
```
you will not take me unsnakecase satan

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -1,7 +1,7 @@
 /datum/action/xeno_action/onclick/lurker_invisibility/use_ability(atom/A)
-	var/mob/living/carbon/Xenomorph/xeno = owner
+	var/mob/living/carbon/Xenomorph/Xeno = owner
```
le snake case

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -69,33 +67,188 @@
+	shake_camera(Target, 2, 1)
+	apply_cooldown_override(40)
+
+/datum/action/xeno_action/activable/flurry/use_ability(atom/A) //flurry ability
```
targeted_atom or w/e

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -69,33 +67,188 @@
+	apply_cooldown_override(40)
+
+/datum/action/xeno_action/activable/flurry/use_ability(atom/A) //flurry ability
+	var/mob/living/carbon/Xenomorph/Xeno = owner
```
snake case...

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -69,33 +67,188 @@
+
+	// Code to get a 1x3 area of turfs
+	var/turf/root = get_turf(Xeno)
+	var/facing = get_dir(Xeno, A)
```
uhh as above with changing A naming

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -69,33 +67,188 @@
+	if (!(!infront || infront.density))
+		temp_turfs += infront_right
+
+	for (var/turf/T in temp_turfs)
```
current_turf or w/e

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -69,33 +67,188 @@
+		target_turfs += T
+		telegraph_atom_list += new /obj/effect/xenomorph/xeno_telegraph/red(T, 2)
+
+	for (var/turf/T in target_turfs)
```
current_turf or w/e

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -69,33 +67,188 @@
+		telegraph_atom_list += new /obj/effect/xenomorph/xeno_telegraph/red(T, 2)
+
+	for (var/turf/T in target_turfs)
+		for (var/mob/living/carbon/Target in T)
```
snake case gamer

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -69,33 +67,188 @@
+			..()
+			return
+
+/datum/action/xeno_action/activable/tail_Jab/use_ability(atom/A)
```
targeted_atom or w/e

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -69,33 +67,188 @@
+
+/datum/action/xeno_action/activable/tail_Jab/use_ability(atom/A)
+	var/mob/living/carbon/Xenomorph/Xeno = owner
+	var/mob/living/carbon/Target = A
```
change to targeted_atom here

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -69,33 +67,188 @@
+		if(barrier != path_turf)
+			to_chat(Xeno, SPAN_WARNING("There's something blocking you from stricking!"))
+			return
+		for(var/obj/structure/S in path_turf)
```
current_structure

harryob commented on code/modules/mob/living/carbon/xenomorph/mutators/strains/lurker/vampire.dm:
```diff
@@ -0,0 +1,38 @@
+
+/datum/xeno_mutator/Vampire/apply_mutator(var/datum/mutator_set/individual_mutators/MS)
+	. = ..()
+	if (. == 0)
```
false?

harryob reviewed (changes requested):
sorry gamer

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -98,4 +95,160 @@
+		if(target.stat == DEAD)
+			return
+		to_chat(xeno, SPAN_XENOHIGHDANGER("You brutally pierce [target] chest with your tail."))
+		xeno.visible_message(SPAN_DANGER("[xeno] pierces [target] chest with his tail."))
```
their

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -98,4 +95,160 @@
+			return
+		if(xeno.action_busy)
+			return
+		xeno.visible_message(SPAN_DANGER("[xeno] Prepares for devastating attack on [target]."), \
```
random caps!!

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -98,4 +95,160 @@
+		if(xeno.action_busy)
+			return
+		xeno.visible_message(SPAN_DANGER("[xeno] Prepares for devastating attack on [target]."), \
+		SPAN_XENOWARNING("You carefully aim your tail towards [target] vital organs."), null, 5)
```
you need an apostrophe here

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -98,4 +95,160 @@
+		xeno.xeno_jitter(1 SECONDS)
+		xeno.flick_heal_overlay(3 SECONDS, "#00B800")
+		target.last_damage_data = create_cause_data(xeno.caste_type, xeno)
+		log_attack("[key_name(xeno)] Executed [key_name(target)] with Tail Jab")
```
```suggestion
		log_attack("[key_name(xeno)] executed [key_name(target)] with Tail Jab")
```

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_abilities.dm:
```diff
@@ -73,3 +73,39 @@
+	action_type = XENO_ACTION_CLICK
+	xeno_cooldown = 30
+
+
```
```suggestion
```

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_abilities.dm:
```diff
@@ -73,3 +73,39 @@
+	xeno_cooldown = 30
+
+
+/datum/action/xeno_action/activable/tail_Jab
```
```suggestion
/datum/action/xeno_action/activable/tail_jab
```

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -98,4 +95,160 @@
+					return
+				target_window.shatter_window(TRUE)
+				apply_cooldown(cooldown_modifier = 0.5)
+				xeno.visible_message(SPAN_XENOWARNING("\The [xeno] strikes the window with his tail!"), SPAN_XENOWARNING("You strike the window with your tail!"))
```
```suggestion
				xeno.visible_message(SPAN_XENOWARNING("\The [xeno] strikes the window with their tail!"), SPAN_XENOWARNING("You strike the window with your tail!"))
```
xenos are girlbosses. or its i guess idc

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -98,4 +95,160 @@
+		if(xeno.action_busy)
+			return
+		xeno.visible_message(SPAN_DANGER("[xeno] prepares for devastating attack on [target]."), \
+		SPAN_XENOWARNING("You carefully aim your tail towards [target]’s vital organs."), null, 5)
```
null arguments >:(
```suggestion
		SPAN_XENOWARNING("You carefully aim your tail towards [target]’s vital organs."), max_distance = 5)
```

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -98,4 +95,160 @@
+			xeno.flick_heal_overlay(1 SECONDS, "#00B800")
+			xeno.gain_health(30)
+			..()
+			return
```
```suggestion
```
this early return means it'll just pick a random human in the 1x3 area. if this is intended just undo this but your PR desc talks about "target**s**"

harryob commented on code/modules/mob/living/carbon/xenomorph/mutators/strains/lurker/vampire.dm:
```diff
@@ -0,0 +1,38 @@
+	if (. == FALSE)
+		return
+
+	var/mob/living/carbon/Xenomorph/Lurker/xeno_client = selected_xeno
```
```suggestion
	var/mob/living/carbon/Xenomorph/Lurker/xeno_client = selected_xeno.xeno
```
runtime (probably my fault)

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -98,4 +95,159 @@
+			log_attack("[key_name(xeno)] attacked [key_name(target)] with Flurry")
+			target.apply_armoured_damage(30, ARMOR_MELEE, BRUTE)
+			playsound(get_turf(target), 'sound/weapons/alien_claw_flesh4.ogg', 30, TRUE)
+			xeno.emote("roar")
```
le epic x3 roar spam
put it outside of the for loop

Canohaaaaaaa commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -98,4 +95,159 @@
+		playsound(xeno, 'sound/effects/alien_tail_swipe1.ogg', 50, TRUE)
+		return
+
+	if(distance > 2)
+		return
```
That check should come earlier, allows you to break windows, from any range given you have LoS.

Diegoflores31 replied:
Clearly intended window breaking telekinesys 3:)

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -98,4 +95,159 @@
+		playsound(xeno, 'sound/effects/alien_tail_swipe1.ogg', 50, TRUE)
+		return
+
+	if(distance > 2)
+		return
```
above user is correct move this early return diego...

Diegoflores31 replied:
fixed . i will upload the fix soon

Diegoflores31 commented:
**update** 
-replace var names on /mutator to match the others
-flurry now actually applies AOE
-Tail jab now correctly executes downed marines
-fixed road issues
- no more telekinetic window breaking tail jab #1984

morrowwolf reviewed (changes requested):
Tailjab breaking windows should have a sound.

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -98,4 +95,160 @@
+				var/obj/structure/window/framed/target_window = current_structure
+				if(target_window.unslashable)
+					return
+				target_window.shatter_window(TRUE)
+				apply_cooldown(cooldown_modifier = 0.5)
+				xeno.visible_message(SPAN_XENOWARNING("\The [xeno] strikes the window with their tail!"), SPAN_XENOWARNING("You strike the window with your tail!"))
+				playsound(get_turf(target_window),'sound/effects/glassbreak3.ogg', 30, TRUE)
```
```suggestion
				playsound(get_turf(target_window),'sound/effects/glassbreak3.ogg', 30, TRUE)
				target_window.shatter_window(TRUE)
				apply_cooldown(cooldown_modifier = 0.5)
				xeno.visible_message(SPAN_XENOWARNING("\The [xeno] strikes the window with their tail!"), SPAN_XENOWARNING("You strike the window with your tail!"))
```
mr flores move this proc! (so the sound actually happens)

Diegoflores31 commented:
**update ultimate ultra** 
changed execution text to be more soul "Head bite"
removed tail stab from vamp lurker ( not intended in the first place since tail jab was done first)
adds sway effect when Rushing on someone
changed DAZED effect to actually use the proc

increased cooldown on failed tail stab . 
increased damage of tail stab to 30

harryob commented on code/modules/mob/living/carbon/xenomorph/abilities/lurker/lurker_powers.dm:
```diff
@@ -98,4 +95,161 @@
+			return
+		if(xeno.action_busy)
+			return
+		xeno.visible_message(SPAN_DANGER("[xeno] grabs [target]’s head agressively."), \
+		SPAN_XENOWARNING("You grab [target]’s head aggresively."), max_distance = 5)
```
```suggestion
		xeno.visible_message(SPAN_DANGER("[xeno] grabs [target]’s head aggressively."), \
		SPAN_XENOWARNING("You grab [target]’s head aggressively."), max_distance = 5)
```
